client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -147,167 +168,225 @@ export class DisplayTable extends React.Component {
     );
     const is_total_exist = !_.isEmpty(total_row);
 
-    const ordered_column_keys = _.chain(col_configs_with_defaults)
+    const all_ordered_col_keys = _.chain(col_configs_with_defaults)
       .map(({ index }, key) => [index, key])
       .sortBy(_.first)
       .map(_.last)
       .value();
-    const csv_string = _.chain(ordered_column_keys)
+    const visible_ordered_col_keys = _.intersection(
+      all_ordered_col_keys,
+      visible_col_keys
+    );
+    const csv_string = _.chain(visible_ordered_col_keys)
       .map((key) => col_configs_with_defaults[key].header)
       .thru((header_row) => [header_row])
       .concat(
         _.map(sorted_filtered_data, (row) =>
-          _.map(ordered_column_keys, (key) =>
+          _.map(visible_ordered_col_keys, (key) =>
             col_configs_with_defaults[key].raw_formatter(row[key])
           )
         )
       )
       .thru((csv_data) => d3.csvFormatRows(csv_data))
       .value();
 
+    const util_components_default = {
+      copyCsvUtil: (
+        <DisplayTableCopyCsv key="copyCsvUtil" csv_string={csv_string} />
+      ),
+      downloadCsvUtil: (
+        <DisplayTableDownloadCsv
+          key="downloadCsvUtil"
+          csv_string={csv_string}
+          table_name={table_name}
+        />
+      ),
+      columnToggleUtil: (
+        <DisplayTableColumnToggle
+          key="columnToggleUtil"
+          columns={
+            <LegendList
+              items={_.map(all_ordered_col_keys, (key) => ({
+                id: key,
+                label: col_configs_with_defaults[key].header,
+                active: _.includes(visible_ordered_col_keys, key),
+              }))}
+              onClick={(clicked_key) => {
+                // 1st column cannot be toggled off
+                !(visible_ordered_col_keys[0] === clicked_key) &&
+                  this.setState({
+                    visible_col_keys: _.toggle_list(
+                      visible_col_keys,
+                      clicked_key
+                    ),
+                  });
+              }}
+            />
+          }
+        />
+      ),
+    };
+    const util_components_with_defaults = _.chain({
+      ...util_components_default,
+      ...util_components,
+    })
+      .map()
+      .reverse()
+      .value();
```

Comment:
I still have opinions about the implementation of this, but we went back and forth on it enough already, haha. Fine for now. If you still want to know how I was thinking of doing it let me know next week and I'll make a quick draft!

Reply:
Huh, I thought this was kinda what we both had in mind😂, default utils that can be changed, and within the same prop, can add any new custom utils. Only using one object prop.. Feel free to make changes !